### PR TITLE
V2 linter API support

### LINKED ISSUE #6: Support v2 Linter API
In the [v2.3.0 release](https://github.com/steelbrain/linter/releases/tag/v2.3.0) of `linter` support for the v1 API was removed, breaking semver. This means that this provider is no longer supported by the current version of `linter`.

To get this provider working again it will need to be upgrade to the v2 API, a guide for accomplishing that can be found here: [Upgrading to Standard Linter v2](https://github.com/steelbrain/linter/blob/v2.3.0/docs/guides/upgrading-to-standard-linter-v2.md).

Until this is fixed in order to use this provider you will need to downgrade to the last version of `linter` that supports the v1 API, you can do this with `apm install linter@2.2.0`.

### PR DESCRIPTION
**Pull request by vzamanillo**
Fixes #6

### CODE REVIEW

**Comment by vzamanillo** [comment]
https://stackoverflow.com/questions/39937224/travis-go-error-the-command-eval-go-get-t-v-failed

**Comment by Arcanemagus** [comment]
You'll want to add this to the `travis.yml` to tell it not to attempt the default:
```yml
install: skip
```

**Comment by Arcanemagus** [comment]
Enabled Circle CI so the next commit should trigger a build.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  working_directory: /tmp/project
+  docker:
+    - image: arcanemagus/atom-docker-ci:stable
+  steps:
+    # Restore project state
+    - attach_workspace:
+        at: /tmp
+    - run:
+        name: Install Go
+        # Note, using ruby-full instead of ruby for the headers to compile rubocop
```
Comment (Arcanemagus) [inline .circleci/config.yml]: 🔥 Doesn't mean anything here.

File: package.json [hunk 2]
```diff
@@ -20,5 +21,78 @@
   "license": "MIT",
   "engines": {
     "atom": ">=0.174.0 <2.0.0"
```
Comment (Arcanemagus) [inline package.json]:
```suggestion
    "atom": ">=1.9.0 <2.0.0"
```

File: package.json [hunk 3]
```diff
@@ -20,5 +21,78 @@
...
+  "configSchema": {
+    "executablePath": {
+      "title": "Go Executable Path",
+      "type": "string",
+      "description": "The path to the `go` executable. Find by running `which go`",
+      "default": "go"
+    },
+    "extraOptions": {
+      "type": "string",
+      "default": "-v",
```
Comment (Arcanemagus) [inline package.json]: Is this `-v` _required_ for the linter to work? If so it shouldn't be left up to the user to leave it in there, it should be added later no matter what in the code.

Comment (vzamanillo) [reply package.json]: I'd  preserved it during the migration from Coffe but I I thought the same thing as you.

**Comment by vzamanillo** [comment]
> Enabled Circle CI so the next commit should trigger a build.

Circle CI was not triggered.

**Comment by Arcanemagus** [comment]
Whoops, forgot to enable builds for forks, should be good now.

File: package.json (hunk 3 above)
Comment (Arcanemagus) [reply package.json]: Ah, so it looks like that enables additional output. The config you added for this works just fine!

**Comment by Arcanemagus** [comment]
Managed to trigger CircleCI without needing a commit done here, and it looks like that's passing as well.

**Pull request merged**